src/arithmetic/algebraic_expression.c

```diff
@@ -16,94 +18,84 @@ AlgebraicExpression *_AE_MUL(size_t operand_cap) {
     ae->operand_count = 0;
     ae->operands = malloc(sizeof(AlgebraicExpressionOperand) * ae->operand_cap);
     ae->edge = NULL;
-    ae->edgeLength = NULL;
     return ae;
 }
 
-int _intermidate_node(const Node *n) {
+static int _intermidate_node(const Node *n) {
     /* ->()<- 
      * <-()->
      * ->()->
      * <-()<- */
-    return ((Vector_Size(n->incoming_edges) > 1) ||
-            (Vector_Size(n->outgoing_edges) > 1) ||
-            ((Vector_Size(n->incoming_edges) > 0) && (Vector_Size(n->outgoing_edges) > 0)));
+    return ((array_len(n->incoming_edges) > 1) ||
+            (array_len(n->outgoing_edges) > 1) ||
+            ((array_len(n->incoming_edges) > 0) && (array_len(n->outgoing_edges) > 0)));
```

Comment:
I think `(array_len(n->incoming_edges) + array_len(n->outgoing_edges)) >= 2` would be a clearer condition.

Reply:
👍